src/app/clusters/testing/CommandTesting.h

```diff
@@ -0,0 +1,269 @@
+/*
+ *    Copyright (c) 2025 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <app/CommandHandler.h>
+#include <app/ConcreteCommandPath.h>
+#include <app/MessageDef/CommandDataIB.h>
+#include <app/data-model-provider/ActionReturnStatus.h>
+#include <app/data-model-provider/OperationTypes.h>
+#include <lib/core/CHIPError.h>
+#include <lib/core/DataModelTypes.h>
+#include <lib/core/TLVReader.h>
+#include <lib/core/TLVWriter.h>
+#include <lib/support/CodeUtils.h>
+#include <messaging/ExchangeContext.h>
+
+#include <memory>
+#include <vector>
+
+namespace chip {
+namespace app {
+namespace Testing {
+
+// Mock CommandHandler for testing command invocations
+class MockCommandHandler : public CommandHandler
```

Comment:
https://github.com/project-chip/connectedhomeip/pull/41508 added a ClusterTester.

Would it make sense to include some command testing ability in that code?

I think we want code to read like `tester.InvokeCommand(id, request, response)` and get back a response. Where request and response are optional (if there are no structures for it)